### ISSUE TITLE
.groupBy.aggregate only returns the first grouping

### ISSUE THREAD

**Issue by sw-yx** [data-forge/data-forge-js#2]
Hi Ashley, first of all I really admire what you are trying to do with data-forge! here are issues I encountered when trying to use it.

first off, your canonical example for groupby+aggregate seems broken:

```
// Group by client.
var summarized = salesData
    .groupBy(row => row.ClientName);
    .select(group => ({
        ClientName: group.first().ClientName,

        // ERROR: deflate is not a function of group
        Amount: group.deflate(row => row.Sales).sum(),
    }))
    .inflate() // Series -> dataframe.
    .toArray(); // Convert to regular JS array.
```

i havent dug through your various versions but it seems that you may not have implemented `deflate` yet? the example doesnt even actually use `aggregate`.

now when i try to roll my own aggregation:

```
var summary = dataFrame
            .groupBy(row => row.account) // i know there are multiple groups
            // // .skip(1)
            .aggregate(seed, 
                (agg, row) => { //only returns the first group
                    return {
                        account: row.toArray().map(x=> x.account), 
                        series: row.toArray().map((x)=>{return {date:x.date, count:x.actcount}}),
                    };
                }
            );
```

just to be clear, what this code is supposed to be doing is groupBy `account` name, and then display a pairing of unique account name, alongside a dataframe of corresponding `date`s and `actcount`s for each `account`. When i run this I only get a result for the first matching group and then the other results are not returned, which makes me think i need some sort of `map` somewhere but I can't figure out where to put it!

am i missing some `forEach` or `map` somewhere? i cant figure out how to implement. I'm relatively new to node but have messed around with data wrangling in d3.

**Comment by sw-yx** [comment]
i have done a hacky ugly workaround that returns the desired data per group:

```
        var summary = dataFrame
            .groupBy(row => row.account)
            .toPairs();
        summary = summary.map((x)=> {return {account:x[0],data:x[1].toPairs()}})
```

but obviously this is doing what the `aggregate` function should have done out of the box

**Comment by ashleydavis** [comment]
There were some issues with that code snippet. 

The call to deflate is not correct because the group is already a series. I'm rethinking the design of inflate and deflate which is used to convert a series to a dataframe and vice-versa. In the meantime I've changed the code snippet to use `select` instead of `deflate`.

I've also added an expanded working example for this code snippet to the [Data-Forge examples repo](https://github.com/data-forge/data-forge-js-examples-and-tests). Look for example 13.

Thanks for your feedback.
